examples/launch_templates/main.tf

```diff
@@ -23,6 +23,23 @@ provider "template" {
   version = "~> 2.1"
 }
 
+data "aws_eks_cluster" "cluster" {
+  name = module.eks.cluster_id
+}
+
+data "aws_eks_cluster_auth" "cluster" {
```

Comment:
It seems there is inconsistency in the usage of the `aws_eks_cluster_auth`.
The first time runs perfectly, but the subsequents run it fails with `unathorized`.
My guess here is that once Terraform stored the token data in `data`, is already expired for the next runs. I couldn't find a way to force the token update on each run.
I've opened this issue: https://github.com/terraform-providers/terraform-provider-aws/issues/10362

Reply:
Are you sure that it is not working in that context? We faced the same problem and it was caused by the depends_on in the "aws_eks_cluster_auth" - which is present in your sample, but not this code.

If the dependency is induced by referencing the output of the module it seems to work.

Reply:
The problem wasn't the `depends_on` it was that the cluster name must be got from the resource  *aws_eks_cluster*, and not from a var or local.